### ISSUE TITLE
"Username" instead "Name" as sender in mail

### ISSUE THREAD

**Issue by emikolajczak** [RocketChat/Rocket.Chat#15125]
### Description:

When user received an e-mail i.e. about new message, as sender is being shown "Username". In old versions there was an "Name". It is confusing for users.

### Steps to reproduce:

1. Receive an e-mail i.e. about new message
2. "Username" (login) show instead "Name" in sender.

### Expected behavior:

"Name" is being shown as sender. Second solution is make possible to choose "Username" or "Name" should be shown.

### Actual behavior:

"Username"is being shown as sender.

### Server Setup Information:

Version of Rocket.Chat Server: 1.2.1
Operating System: Centos 7
Deployment Method: docker compose
Number of Running Instances: 9
DB Replicaset Oplog: yes
NodeJS Version: 8.11.4 - x64
MongoDB Version: 4.0

**Comment by reetp** [comment]
If nothing else I'd upgrade to at least 1.2.3 soonest if I was you....

You should also always test on latest as per this please:

https://rocket.chat/docs/contributing/reporting-issues

However, I have had a look myself and can see thatis occurs on v1.3.0

I can see this changed around May time so possibly introduced with either v1.0.x or v1.1.x which makes this a regression to previous behaviour.

Previous behaviour:

> Rocket.Chat
> Joe Blogs sent a message on #ourchannel:

Current behaviour (with no changes in Administration)

> Rocket.Chat
> joe sent a message on #ourchannel:

I think the file responsible is probably here but my JS skills are non existent to be able to see why this occurs.

https://github.com/RocketChat/Rocket.Chat/blob/master/app/lib/server/functions/notifications/email.js

This seems to suggest it should use RealName if set otherwise use username?

```
export function sendEmail({ message, user, subscription, room, emailAddress, hasMentionToUser }) {
	const username = settings.get('UI_Use_Real_Name') ? message.u.name || message.u.username : message.u.username;
```

Mine is set to Layout Interface use Real Name but it still gets UserName

I agree with the OP though - this should be configurable in admin/Email Subject

So you should be able to specify :

[UserName] or  [RealName]

**Comment by Paionir** [comment]
Same problem here:

Rocket.Chat Version | 3.0.4
Apps Engine Version |  1.12.0
OS Type | Linux Centos 7.7
OS Release | 3.10.0-1062.1.2.el7.x86_64
Node | v12.16.1
MongoDB | 4.0.6
Mongo Storage Engine | mmapv1


I tried installing a develop version using meteor and editing this line in the file https://github.com/RocketChat/Rocket.Chat/blob/master/app/lib/server/functions/notifications/email.js :
```
const userName = s.escapeHTML(settings.get('UI_Use_Real_Name') ? message.u.name || message.u.username : message.u.username);
```
to:
```
const userName = s.escapeHTML(settings.get('UI_Use_Real_Name') ? message.u.name : message.u.username);
```

With this edit I get neither the username, nor the real name in the mail body, but just an empty string. 
Since settings value "UI_User_Real_Name" is set to true, this means that **message.u.name** is not set.

I checked in the mongodb user collection in rocketchat db and the "name" field is correctly set as the "username" field.

That is how far I can go with my analysis. Hope it helps

**Comment by dougfabris** [comment]
Hi there, thanks for the contribution! 🚀 💯 

Sorry, this error does not happen on the latest and it was fixed here: #22309

[Rocket.Chat](https://rocket.chat/)  provides support for, at least, 3 months for each release and 6 months for the latest releases of each major version

I would suggest you update your Rocket.Chat version to the latest stable version, which you can find in [Releases Page](https://github.com/RocketChat/Rocket.Chat/releases) and, check it [HERE](https://docs.rocket.chat/resources/get-support/enterprise-support#rocket.chat-versions) the table with versions and End of Life dates. 

---

**Questions? Help needed? Feature Requests?** 

- Join our [Open Server](https://open.rocket.chat/) in the #support channel and feel free to raise a question
- Join our [Community Forum](https://forums.rocket.chat/) and search/create a post there
